### ISSUE TITLE
Add power to landuse, landuse-labels

### ISSUE THREAD

**Issue by nvkelso** [mapzen/vector-datasource#186]
We're missing polygons and labels for large power generating stations.

https://taginfo.openstreetmap.org/keys/power?filter=relations#values

These three values seem to be the useful ones:

- [x] plant
- [x] generator
- [x] substation

For these 2 deprecated values, let's recast them into `substation` landuse kind.

- [x] station
- [x] sub_station

Places to change:

* https://github.com/mapzen/vector-datasource/blob/master/queries/landuse.jinja2

* https://github.com/mapzen/vector-datasource/blob/master/queries/landuse_labels.jinja2

* https://github.com/mapzen/vector-datasource/blob/master/data/functions.sql

* https://github.com/mapzen/TileStache/blob/integration-1/TileStache/Goodies/VecTiles/transform.py#L592

And maybe somewhere else, do we need new indexes? @rmarianski for comment.

**Comment by rmarianski** [comment]
> do we need new indexes?

Doesn't look like we'll need new indexes, but the `mz_calculate_is_landuse` function will need to start accepting the `power` tag.

**Comment by zerebubuth** [comment]
@nvkelso, what's an appropriate order for power "landuse" polygons? the same as commercial / hospital / etc... (2)?

@rmarianski, for folding in the station/sub_station values - you'd prefer that done in Python rather than the SQL, right?

**Comment by rmarianski** [comment]
> for folding in the station/sub_station values - you'd prefer that done in Python rather than the SQL, right?

If it's just a `coalesce`, then I think doing it in sql is fine, and some of the existing queries do it there too. If the logic is more complex though, imho it's easier to maintain in python.

**Comment by nvkelso** [comment]
Are we exporting the calculated landuse sort key now? That should have been included in last the road intercut work, but I don't see it.

**Comment by zerebubuth** [comment]
> If it's just a coalesce, then I think doing it in sql is fine, and some of the existing queries do it there too. If the logic is more complex though, imho it's easier to maintain in python.

Sadly not, it would be `CASE WHEN ... END`, so I'll do it in Python. The only downside to that is that if we end up having more than one value "station" then we'll have lost the link back to the "power" key. But we can cross that bridge when we get to it.

**Comment by nvkelso** [comment]
> @nvkelso, what's an appropriate order for power "landuse" polygons? the same as commercial / hospital / etc... (2)?

In this section:
https://github.com/mapzen/TileStache/blob/integration-1/TileStache/Goodies/VecTiles/transform.py#L688-L717

We generally default to `1` except for features we know need to stack on top of another larger landuse. If you look at the Presideo in San Francisco you'll see both a golf course and a cemetery inside of the large park. So `park` gets `1` and `golf_course` and `cemetery` both get sort order of `2`. 

For `power` variants of landuse, can we assume they are always at the bottom of the stacking order? I think generally speaking maybe, but it seems like something like a `substation` wouldn't be cut out of an enclosing landuse like `park` or `urban`.

**Comment by zerebubuth** [comment]
> For power variants of landuse, can we assume they are always at the bottom of the stacking order? I think generally speaking maybe, but it seems like something like a substation wouldn't be cut out of an enclosing landuse like park or urban.

Sounds like you think this would be appropriate:

```yaml
plant: 1
generator: 1
substation: 2
```
?

**Comment by nvkelso** [comment]
In that direction, yeah. Expanding on this thread, something like this is probably more right: 

kind	orig	new	alpha_sort_orig	alpha_sort_new
    'aerodrome'	2	4	1	1
    'apron'	3	5	2	2
    'cemetery'	2	4	3	3
    'commercial'	2	4	4	4
    'conservation'	1	2	5	5
    'farm'	1	3	6	6
    'farmland'	1	3	7	7
    'forest'	1	3	8	8
    'generator'		3	29	9
    'golf_course'	2	4	9	10
    'hospital'	2	4	10	11
    'nature_reserve'	1	2	11	12
    'park'	1	2	12	13
    'parking'	2	4	13	14
    'pedestrian'	2	4	14	15
    'place_of_worship'	2	4	15	16
    'plant'		3	28	17
    'playground'	2	4	16	18
    'railway'	2	4	17	19
    'recreation_ground'	1	4	18	20
    'residential'	1	1	19	21
    'retail'	2	4	20	22
    'runway'	3	5	21	23
    'rural'	1	1	22	24
    'school'	2	4	23	25
    'stadium'	1	3	24	26
    'substation'		4	30	27
    'university'	2	4	25	28
    'urban'	1	1	26	29
    'zoo'	2	4	27	30

**Comment by nvkelso** [comment]
I'm seeing features like 'meadow' in the road intercut landuse_kind values, what do things default to if they aren't listed in the sort function?

**Comment by zerebubuth** [comment]
I'm not sure that's hooked up correctly anyway. And it won't work at all if you're not seeing an 'order' element on the "landuse" polygons... I think this should go in as a separate issue.

**Comment by nvkelso** [comment]
Something like: https://github.com/mapzen/TileStache/commit/6624ec996919dc31cce1337bcb66e0275930b702

**Comment by zerebubuth** [comment]
Before:

![power_landuse_before](https://cloud.githubusercontent.com/assets/271360/9820395/9cfc62fa-58ab-11e5-9acf-d04075c8d53c.png)

After #199:

![power_landuse_after](https://cloud.githubusercontent.com/assets/271360/9820397/a4bd171e-58ab-11e5-8d71-fc276c85c2b6.png)

Using this in the `scene.yaml`:

```yaml
    landuse:
        data: { source: osm }
        draw:
            polygons:
                order: 1
                color: '#aaffaa'
        power:
          filter: { kind: [plant, generator, substation] }
          draw:
            polygons:
              order: 2
              color: '#801010'
```

**Comment by nvkelso** [comment]
All related PR have been merged.
